### DISCUSSION TITLE
предложения - ChatService

### DISCUSSION THREAD

**Discussion by justqstn** [Sokol-Games-GmbH/PixelCombats.PublicApi#95] [Ideas]
сервис работы с чатом
прошу реализовать это первее всего, ведь с помощью этого сервиса можно будет делать многое и упростить еще больше

[ScriptingRoot("Chat")]

методы, свойства, события:

void Mute(); // отключает чат для всех игроков
void UnMute(); // включает возможность писать в чат. в аргументах 
Chat.GetContext().Mute();
Chat.GetContext(Teams.Get("Red")).Mute();
мут работает таким образом - у игрока появляется иконка на чате о том, что он замьючен (к примеру это будет тоже перекрестие), и его сообщения не приходят другим игрокам, но событие ниже будет срабатывать при отправке сообщеий

void SendMessage(string message);  // отправит сообщение в чат, кому именно зависит от контекста
Chat.GetContext().SendMessage("Hello"); 
Предлагаю разрешить использовать в этих сообщениях unitymarkup, потому что в обычном чате он не работает

ApiEvent<ApiPlayer Player, string Message> OnMessage(); // когда в чате появилось сообщение. в аргументы передает того, кто отправил сообщение, и само сообщение
Chat.OnMessage.Add(function(plr, msg) {
    if (plr.IdInRoom == 1 && msg == "/restart") Game.RestartGame();
    // когда игрок с рум айди 1 отправит в чат /restart, то произойдет рестарт комнаты
});

**Comment by kkohno** [comment]
я понял, что нужна какаято консолька, чтобы слать команды в скрипты, сделаем... скорее всего любое сообщение, которое начинается со / будут считаться командами и не выводиться. зато ловиться скриптами... тут продумывать нужно

**Comment by justqstn** [comment]
при желании можно и консоль, и подобную шнягу сделать, потому что мут каких-то игроков лишним не будет. к примеру спамеры.
надо будет их как-то помечать в лидерборде
